sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -228,6 +231,24 @@ public EventHubProducerClient(string connectionString) : this(connectionString,
             Connection = connection;
             RetryPolicy = clientOptions.RetryOptions.ToRetryPolicy();
             EventHubProducer = Connection.CreateTransportProducer(null, RetryPolicy);
+            PartitionProducerPool = new TransportProducerPool();
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubProducerClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="connection">The connection to use as the basis for delegation of client-type operations.</param>
+        /// <param name="transportProducer">The transport producer instance to use as the basis for service communication.</param>
+        ///
+        /// <remarks>
+        ///   This constructor is intended to be used internally for functional
+        ///   testing only.
+        /// </remarks>
+        ///
+        internal EventHubProducerClient(EventHubConnection connection,
+                                        TransportProducer transportProducer): this(connection, transportProducer, new TransportProducerPool())
```

Comment:
I don't think we need this overload.   We could just make the pool optional and default to null.   In the constructor, if it is null, then new up an instance,  no?